src/components/calcite-inline-editable/calcite-inline-editable.tsx

```diff
@@ -0,0 +1,265 @@
+  @Prop({ reflect: true }) hasControls = false;
+
+  /** specify text to be user for the enable editing button's aria-label, defaults to `Click to edit` */
+  @Prop({ reflect: true }) intlEnableEditing = TEXT.intlEnablingEditing;
+
+  /** specify text to be user for the cancel editing button's aria-label, defaults to `Cancel` */
+  @Prop({ reflect: true }) intlCancelEditing = TEXT.intlCancelEditing;
+
+  /** specify text to be user for the confirm changes button's aria-label, defaults to `Save` */
+  @Prop({ reflect: true }) intlConfirmChanges = TEXT.intlConfirmChanges;
+
+  /** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the scale of the closest wrapping component with a set scale */
+  @Prop({ reflect: true }) scale?: "s" | "m" | "l";
+
+  /** specify the scale of the inline-editable component, defaults to the scale of the wrapped calcite-input or the theme of the closest wrapping component with a set theme */
+  @Prop({ reflect: true }) theme?: "light" | "dark";
+
+  //** when has-controls, specify a callback to be executed prior to disabling editing. when provided, loading state will be handled automatically. */
+  @Prop() onConfirmChanges: () => Promise<any>; // eslint-disable-line @stencil/decorators-style
+
+  //--------------------------------------------------------------------------
+  //
+  //  Getters/Setters
+  //
+  //--------------------------------------------------------------------------
+
+  get shouldShowControls(): boolean {
+    return this.editingEnabled && this.hasControls;
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  //--------------------------------------------------------------------------
+
+  componentWillLoad() {
+    this.inputElement = this.el.querySelector("calcite-input") as HTMLCalciteInputElement;
+    this.scale =
+      this.scale || this.inputElement.scale || getElementProp(this.el, "scale", undefined);
+    this.theme =
+      this.theme || this.inputElement.theme || getElementProp(this.el, "theme", undefined);
+  }
+
+  componentDidLoad() {
+    this.htmlInput = this.inputElement.querySelector("input");
+    if (!this.editingEnabled) this.htmlInput.tabIndex = -1;
+  }
+
+  render(): VNode {
+    return (
+      <Host>
+        <div
+          class="calcite-inline-editable-wrapper"
+          onClick={this.enableEditingHandler}
+          onKeyDown={this.escapeKeyHandler}
+        >
+          <div class="calcite-inline-editable-input-wrapper">
+            <slot />
+          </div>
+          <div class="calcite-inline-editable-controls-wrapper">
+            {!this.editingEnabled && (
+              <calcite-button
+                appearance="transparent"
+                aria-label={this.intlEnableEditing}
+                class="calcite-inline-editable-enable-editing-button"
+                color="dark"
+                disabled={this.inputElement.disabled}
```

Comment:
note: if inputElement.disabled changes then this wouldn't update automatically. right? Is that an expected behavior?

Reply:
Good catch. Thanks, @driskull. Added a disabled prop to the inline-editable comp and now passing it down to it's child input rather than vice versa.